technical/binary_tree/c++/binary_tree.h

```diff
@@ -0,0 +1,272 @@
+#include <stack>
+#include <iostream>
+
+template <typename T>
+// Expects type T to have an operator overload for <. 
+class binary_tree {
```

Comment:
for custom types in C++, many standards prefer CamelCase for class names, with the first letter capitalized.